src/app/pages/send/send-crypto-asset-form/_components/amount-field.tsx

```diff
@@ -1,54 +1,88 @@
-import { Box, Flex, Stack, Text, color } from '@stacks/ui';
+import { useEffect, useState } from 'react';
+
+import { Flex, Input, Stack, Text, color } from '@stacks/ui';
 import { useField } from 'formik';
 
+import { STX_DECIMALS } from '@shared/constants';
+import { Money } from '@shared/models/money.model';
+
 import { ErrorLabel } from '@app/components/error-label';
 
-import {
-  amountInputId,
-  assetSymbolId,
-  maxInputContainerWidth,
-  useFontResizer,
-} from '../_hooks/use-font-resizer';
+const maxFontSize = 48;
+const minFontSize = 22;
+const maxLengthDefault = STX_DECIMALS + 1; // + 1 for decimal char
 
-interface AmountFieldProps {
+interface GetAmountModifiedFontSize {
+  amount: string;
+  fontSize: number;
+  fontSizeModifier: number;
+  maxFontSize: number;
   symbol: string;
+}
+function getAmountModifiedFontSize(props: GetAmountModifiedFontSize) {
+  const { fontSize, fontSizeModifier, maxFontSize, amount, symbol } = props;
+  const convertedAmountFontSize = amount.length * fontSizeModifier;
+  return amount.length > symbol.length
+    ? Math.ceil(fontSize - convertedAmountFontSize)
+    : maxFontSize;
+}
+
+interface AmountFieldProps {
+  balance: Money;
   rightInputOverlay: JSX.Element;
```

Comment:
I named this badly, as it's actually positioned at the bottom, so better `bottomInputOverlay`